### DISCUSSION TITLE
Twig filter that makes links clickable for text variable

### DISCUSSION THREAD

**Discussion by d47081** [symfony/symfony#52339] [Q&A] [answered]
Hello everyone!

I want to replace all URLs in text to clickable links (cover to the `a` tag) but can't find any filter or macros that's do this.

Something like `{{ 'text http://1 text http://2' | make_all_links_clickable_filter }}`

As the solution, I can write custom twig filter extension, but in this case I must to put html in controller (where links contains template-dependent classes) and apply the `raw` filter in template to output chars like `"`, `<` etc.

Maybe exists some actual solution?

Just also tried to play with `replace` filter, but looks like it not support regex replacements.

Stuck so.

**Comment by sstok** [comment, marked as answer]
There is no filter in Twig that supports this, this is more related to markdown or ubb type formatting.

So you do need a custom filter.

**Reply by d47081** [reply]
Thank you for advice with markdown idea, implemented without raw output and html in controllers

If somebody have same issue, here is my solution:
https://github.com/YGGverse/YGGtracker/commit/6dcbd6de401732501c612433f5b71a861da2c6c9
